src/parser/GenericParser.ts

```diff
@@ -42,69 +42,49 @@ export class GenericParser {
     async calculateMetrics() {
         const fileMetrics = new Map<string, FileMetricResults>();
         const unsupportedFiles: string[] = [];
+        const errorFiles: string[] = [];
         let couplingMetrics = {} as CouplingResult;
 
         try {
             const filePathGenerator = findFilesAsync(this.config);
-            const parsePromises = new Map<string, Promise<SourceFile | null>>();
+            const parsePromises = new Map<string, Promise<SourceFile>>();
```

Comment:
map is not needed anymore (see below)
```suggestion
            const parsePromises: Promise<SourceFile>[] = [];
```